src/mavoscript.js

```diff
@@ -321,7 +321,62 @@ var _ = Mavo.Script = {
 				return ret;
 			},
 			precedence: 3
-		}
+		},
+		"groupby": {
+			symbol: "by",
+			code: (array, key) => {
+				if (Array.isArray(array) && Array.isArray(key))  { 
+					var propName = key[Mavo.as] === true ? key[Mavo.name] : $.value(key[0], Mavo.toNode, "property");
+					var temp = new Mavo.BucketMap({arrays: true});
+					var ret = [];
+					ret[Mavo.groupedBy] = true;
+
+					for (var i = 0; i < array.length; i++) {
+						const k = i < key.length ? Mavo.value(key[i]) : null;
+						temp.set(k, Mavo.value(array[i]));
+					}	
+
+					for (var [value, items] of temp) {
+						var obj = {"$value": value};
+
+						if (propName !== undefined) {
+							obj[propName] = value;
+						}
+						
+						obj["$items"] = items;
+						ret.push(obj);
+					}
+
+					return ret;
+				}
+				else { // invalid arguments, convert to arrays
+					// return (Mavo.toArray(array), Mavo.toArray(key));
+				}
+			},
+			precedence: 2
+		},
+		"as": {
+			symbol: "as",
+			code: (property, name) => {
+				if (property !== undefined && $.type(property) === "array" && name !== undefined) {
+					if ($.type(name) === "string" || $.value(name[0], Mavo.toNode, "property") !== undefined) {
+						var ret = property.slice(0);
+						ret[Mavo.name] = $.type(name) === "string" ? name : $.value(name[0], Mavo.toNode, "property");
+						ret[Mavo.as] = true;
+						return ret;
+					}
+
+					return property;
+				}
+				else if (property !== undefined) {
+					return property;
+				}
+				else {
+					return null;
```

Comment:
No need for `else` here, since all previous clauses return. Just `return null`, by itself, which makes it immediately obvious that it's a fallback.

In fact, you can just `return property` and leave it at that. I don't think there's much use in drilling down into what kind of falsy value exactly is `property`. One fallback for invalid data is sufficient.